core/trino-main/src/main/java/io/trino/execution/scheduler/PipelinedQueryScheduler.java

```diff
@@ -934,6 +940,32 @@ public static DistributedStagesScheduler create(
             return distributedStagesScheduler;
         }
 
+        private static int getEstimatedHashPartitionCount(StageManager stageManager, int maxHashPartitionCount)
+        {
+            List<PartitioningScheme> schemes = stageManager.getStagesInTopologicalOrder().stream()
+                    .map(stage -> stage.getFragment().getPartitioningScheme())
+                    .filter(schema -> !(schema.getPartitioning().getHandle().equals(SINGLE_DISTRIBUTION)
```

Comment:
maybe:
```
!ImmutableSet.of(SINGLE_DISTRIBUTION, COORDINATOR_DISTRIBUTION, SOURCE_DISTRIBUTION).contains(schema.getPartitioning().getHandle())
```
you could allocate (static) variable for `ImmutableSet.of(SINGLE_DISTRIBUTION, COORDINATOR_DISTRIBUTION, SOURCE_DISTRIBUTION)` for better readability